recipes-core/ww-console-image-initramfs-init/files/initramfs-init-script.sh

```diff
@@ -664,34 +655,18 @@ determinebuildversions(){
 
 determineUbootVersions(){
 	mountboot_ro
-
-	dd if=/dev/mmcblk0 of=/uboot.img skip=$uboot_offset bs=1024 count=$uboot_size >> /dev/null 2>&1
+	availableUboot=$(cat /mnt/.boot/u-boot.bin | grep -a "WigWag-U-boot-version_id" | tail -1 | awk '{print $2}')
+	if [[ "$availableUboot" = "" ]]; then
+		availableUboot=0;
+	fi
+	dd if=/dev/mmcblk0 of=/uboot.img seek=8 bs=1024 count=100 >> /dev/null 2>&1
 	currentUboot=$(cat /uboot.img | grep -a "WigWag-U-boot-version_id" | tail -1 | awk '{print $2}')
 	if [[ "$currentUboot" = "" ]]; then
 		currentUboot=0;
 	fi
-	rm /uboot.img
-
-	expandTMPFS
-	availableUboot=0;
-	if [ -f /tmpfs/boot.tar.xz ]; then
-		if tar xJf /tmpfs/boot.tar.xz -C / ./$uboot_bin 2>/dev/null; then
-			availableUboot=$(cat /$uboot_bin | grep -a "WigWag-U-boot-version_id" | tail -1 | awk '{print $2}')
-			if [[ "$availableUboot" = "" ]]; then
-				# no version info available, but a new u-boot is contained in the upgrade data
-				availableUboot=1;
-			fi
-			rm /$uboot_bin
-		else
-			_decho "No new $uboot_bin available in the boot upgrade data."
-			ls -la /tmpfs/*
-			tar tf /tmpfs/boot.tar.xz
-			_decho "Upgrade may fail if the ROT key has changed."
-		fi
-	fi
-
 	_decho "currentUboot='$currentUboot'"
 	_decho "availableUboot='$availableUboot'"
+	rm /uboot.img
```

Comment:
In this new version of the file uboot will now always be removed. Why is that now the correct behavior?

Reply:
@cmonr  Even in the old version it was always being removed, see line 673.